Bonsai UI docs preview (#27)
ShopBonsai/docs: docs/bonsai-ui/checkout/index.mdx

---
title: Bonsai Checkout
description: Learn about the Bonsai Checkout and how to use it.
sidebar_position: 1
---

This document explains how to use the Bonsai Checkout, and guides you through the process of
integrating it with your website.

# Integrations

There are 2 ways to integrate with Bonsai Checkout:

1. `Embedded` - embedding the Bonsai Checkout in your website
2. `Hosted` - redirecting to secure Bonsai Checkout URL

> Each one of these integrations has its own advantages and disadvantages, and you should choose the one that fits your needs.

Author: ```suggestion
Each of these integrations has its advantages and disadvantages, and you should choose the one that fits your needs.
```

| `bonsai`    | `checkout_ID`   | Starts checkout process using provided cart ID               | Yes      |
| `onSuccess` | `yourSuccessFn` | A _name_ of callback function that will be called on success | No       |
| `onError`   | `yourErrorFn`   | A _name_ of callback function that will be called on error   | No       |

:::info Example
https://your.domain.com/checkout?bonsai=checkout_abcdef

https://your.domain.com/checkout?bonsai=checkout_abcdef&onSuccess=yourSuccessFn&onError=yourErrorFn
:::

`bonsai` parameter is the only required parameter, and it should be provided with a `cart id` that you want to checkout.

### Callbacks

Callbacks are optional, and you can use them to obtain `order` information after a successful checkout, or an `error` in case of a failure.
These function must be assigned to `window` object, and they will be called with `order` or `error` object as their first argument.

:::info Example

```javascript
window.yourSuccessFn = function (order) {
  // do something with order
};

window.yourErrorFn = function (error) {
  // do something with error
};
```

:::

:::caution

> It's important that the names of the callback functions match the names provided in the `onSuccess` and `onError` query parameters.

Reviewer: ```suggestion
The callback functions' names must match those provided in the `onSuccess` and `onError` query parameters.
```
